### PR TITLE
refactor: more oop with stores

### PR DESCRIPTION
**Pull request by Lewdcario**
**Please describe the changes this PR makes and why it should be merged:**
Moves around a lot of things such as `guild#createChannel` into `GuildChannelStore#create`. In the process of doing so, this renames the original `DataStore#create` into `DataStore#add` to free up the `create` function name.
 A thorough list of things changed:
* `Guild#createChannel` -> `GuildChannelStore#create`
* `Guild#ban` -> `GuildMemberStore#ban`
* `Guild#unban` -> `GuildMemberStore#unban`
* `Guild#createEmoji` -> `EmojiStore#create`
* `Guild#createRole` -> `RoleStore#create`
* `Guild#pruneMembers` -> `GuildMemberStore#prune`
* `ClientUser#createGuild` -> `GuildStore#create`
* `Message#clearReactions` -> `ReactionStore#removeAll`
* `MessageReaction#remove` -> `ReactionUserStore#remove`

**Semantic versioning classification:**  
- [x] This PR changes the library's interface (methods or parameters added)
  - [x] This PR includes breaking changes (methods removed or renamed, parameters moved or removed)
- [ ] This PR **only** includes non-code changes, like changes to documentation, README, etc.

### CODE REVIEW

**Comment by SunburntRock89** [comment]
I'm up for all of those except the bulk delete tbh

**Comment by hydrabolt** [comment]
bulkDelete only works in text channels, so moving it to message store might be a bit dodgy

**Comment by Lewdcario** [comment]
Good point, failed to realize that. It has been done~

File: src/structures/interfaces/TextBasedChannel.js [hunk 1]
```diff
@@ -244,11 +244,6 @@ class TextBasedChannel {
    * Messages or number of messages to delete
    * @param {boolean} [filterOld=false] Filter messages to remove those which are older than two weeks automatically
    * @returns {Promise<Collection<Snowflake, Message>>} Deleted messages
-   * @example
```
Comment (bdistin) [inline src/structures/interfaces/TextBasedChannel.js]: Why was this example removed?

Comment (KyeNormanGill) [reply src/structures/interfaces/TextBasedChannel.js]: cuz you suck

Comment (Lewdcario) [reply src/structures/interfaces/TextBasedChannel.js]: That was a mess-up while merging the recent master changes. (I do believe I had an example there before), thank you for telling

**Pull request merged**